### PR TITLE
Disable unnecessary transformations and cleaning up API

### PR DESCRIPTION
**Pull request by bettinaheim**
Since we do a pretty heavy breaking change for our transformation setup, I went ahead and also made a pass to clean up the API. 
This PR also disables subtransformations that are not needed.

### CODE REVIEW

File: src/QsCompiler/Tests.Compiler/ClassicalControlTests.fs [hunk 1]
```diff
@@ -9,7 +9,7 @@ open Microsoft.Quantum.QsCompiler
 open Microsoft.Quantum.QsCompiler.CompilationBuilder
 open Microsoft.Quantum.QsCompiler.DataTypes
 open Microsoft.Quantum.QsCompiler.SyntaxTree
-open Microsoft.Quantum.QsCompiler.Transformations.ClassicallyControlledTransformation
```
Comment (ScottCarda-MS) [inline src/QsCompiler/Tests.Compiler/ClassicalControlTests.fs]: Please revert the renaming for the Classically Controlled, Monomorphization, Monomorphization Validation, and Intrinsic Resolution transformation, as well as the changes to the Utility class so that the changes can be placed in a separate PR.

File: src/QsCompiler/DocumentationParser/DocItem.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ internal abstract class DocItem
         protected readonly string replacement;
 
         /// <summary>
-        /// The item's kind, as a string (Utilities.OperationKind, .FunctionKind, or .UdtKind)
+        /// The item's kind, as a string (Utils.OperationKind, .FunctionKind, or .UdtKind)
```
[the remark this answers is not in the document]
Comment (bettinaheim) [reply src/QsCompiler/DocumentationParser/DocItem.cs]: Just to be consistent with the rest of the code base. 
Everything so far is named Utils rather than Utilities.

**Comment by bettinaheim** [comment]
> Please revert the renaming for the Classically Controlled, Monomorphization, Monomorphization Validation, and Intrinsic Resolution transformations, as well as the changes to the Utility class, so that the changes can be placed in a separate PR.

What is the reason to want it a separate PR? There is no benefit to that in terms of resolving merge conflicts, and logically the API clean up belongs here.

File: src/QsCompiler/Transformations/ClassicallyControlledUtils.cs [hunk 3]
```diff
@@ -31,11 +35,11 @@ public static class Utilities
                     : QsNullable<ImmutableArray<ResolvedType>>.Null),
                 typeArgsMapping,
                 resolvedType,
-                new InferredExpressionInformation(false, false),
+                new InferredExpressionInformation(false, false),
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/ClassicallyControlledUtils.cs]: Trailing space here.

File: src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs [hunk 4]
```diff
@@ -9,9 +9,9 @@
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
 
 
-namespace Microsoft.Quantum.QsCompiler.Transformations.IntrinsicResolutionTransformation
+namespace Microsoft.Quantum.QsCompiler.Transformations.IntrinsicResolution
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs]: Should the file be renamed to IntrinsicResolution.cs as well? The naming convention in this folder seems split between files that end in Transformation and those that don't. Since the namespaces are being shortened it could make sense to shorten the filenames too.

File: src/QsCompiler/Transformations/MonomorphizationTransformation.cs [hunk 5]
```diff
@@ -10,15 +10,16 @@
 using Microsoft.Quantum.QsCompiler.SyntaxTokens;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
 using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Microsoft.Quantum.QsCompiler.Transformations.SearchAndReplace;
 
 
-namespace Microsoft.Quantum.QsCompiler.Transformations.MonomorphizationTransformation
+namespace Microsoft.Quantum.QsCompiler.Transformations.Monomorphization
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/MonomorphizationTransformation.cs]: Same suggestion about renaming file to Monomorphization.cs.

File: src/QsCompiler/Transformations/MonomorphizationValidationTransformation.cs [hunk 6]
```diff
@@ -10,23 +10,23 @@
 using Microsoft.Quantum.QsCompiler.Transformations.Core;
 
 
-namespace Microsoft.Quantum.QsCompiler.Transformations.MonomorphizationValidationTransformation
+namespace Microsoft.Quantum.QsCompiler.Transformations.Monomorphization.Validation
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/MonomorphizationValidationTransformation.cs]: Same suggestion about renaming file to MonomorphizationValidation.cs.

File: src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs [hunk 7]
```diff
@@ -9,9 +9,9 @@
...
 {
-    public class IntrinsicResolutionTransformation
+    public class ReplaceWithTargetIntrinsics
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs]: It seems like this class is not actually a transformation. I'm guessing it's a utility class? Could it be made static?

File: src/QsCompiler/Transformations/ClassicallyControlledTransformation.cs [hunk 8]
```diff
@@ -9,21 +9,24 @@
 using Microsoft.Quantum.QsCompiler.SyntaxTokens;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
 using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Microsoft.Quantum.QsCompiler.Transformations.SearchAndReplace;
 
 
-namespace Microsoft.Quantum.QsCompiler.Transformations.ClassicallyControlledTransformation
+namespace Microsoft.Quantum.QsCompiler.Transformations.ClassicallyControlled
```
Comment (marshallsa) [inline src/QsCompiler/Transformations/ClassicallyControlledTransformation.cs]: I sort of would like to rename this file to ClassicallyControlled.cs as well, although in this case the Transformation part provides contrast with ClassicallyControlledUtils.cs.

File: src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs (hunk 4 above)
Comment (bettinaheim) [reply src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs]: My preference would be to rename the files, too, dropping the *Transformation ending. It seems silly to have all files within a folder named Transformations ending in Transformations as well. The reason they were named this way I believe is that there is a RewriteStep folder in the Compiler project that contains files with the same name which implement the IRewriteStep interface invoking these transformations. Personally, I think having both files have the same name is fine.

File: src/QsCompiler/Transformations/MonomorphizationTransformation.cs (hunk 5 above)
Comment (bettinaheim) [reply src/QsCompiler/Transformations/MonomorphizationTransformation.cs]: I'll resolve the rest of the comments, leaving one open until we decide.

File: src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs (hunk 4 above)
Comment (marshallsa) [reply src/QsCompiler/Transformations/IntrinsicResolutionTransformation.cs]: Oh, I see. Yes, I think it's fine to have the same filenames since it should be clear from context which one is the transformation. If we don't do that then for consistency I'd argue that Conjugations.cs, FunctorGeneration.cs, etc. should be renamed to ConjugationsTransformation.cs, FunctorGenerationTransformation.cs, etc. I prefer the shorter names. :)

Then we also have files like BasicTransformation.cs and CodeTransformations.cs, but having files named Basic.cs and Code.cs would probably be confusing.
